sources/models/src/modeled_types/shared.rs

```diff
@@ -230,6 +231,149 @@ mod test_valid_linux_hostname {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// EtcHostsEntries represents a mapping of IP Address to hostname aliases that can apply to those
+/// addresses.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct EtcHostsEntries(
+    // Ordering matters in /etc/hosts, and this setting directly maps to that file and its behavior in glibc.
+    // Repeated IP Addresses have their host aliases merged to a single line, respecting the order as they appear in this list.
+    Vec<(IpAddr, Vec<ValidLinuxHostname>)>,
+);
+
+impl EtcHostsEntries {
+    pub fn iter_merged(&self) -> impl Iterator<Item = (IpAddr, Vec<ValidLinuxHostname>)> {
+        let mut merged = indexmap::IndexMap::with_capacity(self.0.len());
+
+        for (ip_address, aliases) in &self.0 {
+            merged
+                .entry(ip_address.clone())
+                .or_insert(vec![])
+                .append(&mut (aliases.clone()));
+        }
+
+        merged.into_iter()
+    }
+}
+
+#[cfg(test)]
+mod test_etc_hosts_entries {
+    use super::{EtcHostsEntries, ValidLinuxHostname};
+    use std::net::IpAddr;
+
+    #[test]
+    fn test_valid_etc_hosts_entries() {
+        assert!(serde_json::from_str::<EtcHostsEntries>(
```

Comment:
why can't all these tests be part of the same assert? Same question similar tests below.

Reply:
I've tried to logically separate different checks into multiple assert lines here, so if one fails it's easier to reason about what behavior the check was testing for.